### PR TITLE
fixed unicode comparison in navigation

### PR DESCRIPTION
**Pull request by elgatito**
Sometimes navigation gets to a "fanart" comparison and throws an error regarding unicode conversion fault, so we transform all non-unicode fields to unicode (they can be both unicode and non-unicode). 

Checked with cyrillic names and latin, works for me.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Translation

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by scakemyer** [comment]
I don't think converting every item value to unicode is necessary here. Is the switch in notify() also required / causing actual issues or strings to not be translated properly? I'd rather just convert to unicode both `item['label']` and the localized string in the comparison if possible.

**Comment by elgatito** [comment]
notify() was throwing another exception as it expects string, not <Exception> e. 
Updated commit, now only converting labels.

**Comment by scakemyer** [comment]
Doesn't look like you pushed your new commit

**Comment by elgatito** [comment]
It's here:

```
+        for key in ['label', 'label2']:
 +            if not isinstance(item[key], unicode):
 +                item[key] = unicode(item[key], 'utf-8')
```

**Comment by scakemyer** [comment]
Right, I meant just wrapping the comparison's label, ie. `unicode(item['label'], 'utf-8')`

Sorry to be a pain about this but I'd like to see logs of the other exception for notify, since that's been working here and iirc, `e.reason` is specific to `URLError` only, and the second, catch-all exception *is* supposed to be directly wrapped in unicode() to get the unicode representation of errors. So basically, I'm pretty sure the catch-all is correct, and that `getLocalizedLabel` should just be removed from the `URLError` one since we'll never get a `LOCALIZE[...]` tag in there anyway.

**Comment by elgatito** [comment]
Pushed a change.
Reverted changes over notify(). I don't remember how I got there, but probably when tried with unicode in  Kodi profile path and traceback caused a "non-Ascii" error. Let it stay as it is. 
Will see more precisely if I catch it again.

**Comment by scakemyer** [comment]
Not seeing those last changes again.

**Comment by elgatito** [comment]
Right, not seeing neither. Will test and make changes tomorrow.

File: resources/site-packages/quasar/navigation.py [hunk 1]
```diff
@@ -134,12 +134,12 @@ def run(url_suffix=""):
...
             import traceback
             map(log.error, traceback.format_exc().split("\n"))
-            notify(getLocalizedLabel(unicode(e.reason, 'utf-8')), time=7000)
+            notify(e.reason, time=7000)
         return
     except Exception as e:
         import traceback
         map(log.error, traceback.format_exc().split("\n"))
-        notify(getLocalizedLabel(unicode(e, 'utf-8')), time=7000)
+        notify(e, time=7000)
```
Comment (scakemyer) [inline resources/site-packages/quasar/navigation.py]: This needs to go through `getLocalizedLabel`, as I said I'm pretty sure this one is fine and shouldn't be changed.

File: resources/site-packages/quasar/navigation.py [hunk 2]
```diff
@@ -165,6 +165,11 @@ def run(url_suffix=""):
         if item["label2"][0:8] == "LOCALIZE":
             item["label2"] = getLocalizedLabel(item["label2"])
 
+        # Convert labels to unicode
```
Comment (scakemyer) [inline resources/site-packages/quasar/navigation.py]: There's no use at all in converting `label2`, sorry if I wasn't clear enough, but just wrap `item['label']` in `unicode()` on line 181, keep it simple.

**Comment by elgatito** [comment]
Pushed a change.

**Comment by scakemyer** [comment]
Now that's more like it :+1:

**Comment by elgatito** [comment]
I've got the error with notify(). Not sure what triggered it. Probably timeout while urlopen.

http://paste.ubuntu.com/24057779/

**Comment by scakemyer** [comment]
Well... there was no error there before #740 for error strings... We need to be sure it works for both regular error messages and translated unicode errors from `LOCALIZE` tags...

**Comment by elgatito** [comment]
That is the unicode() causing error. It expects string, while "e" is not always string. 
Why unicode was spaced there - traceback did have unicode characters when profile path had unicode characters.

I will try with 0.9.74 to find a way that suits everyone.

**Comment by scakemyer** [comment]
Think it should just be `getLocalizedLabel(e)` really...

**Pull request merged**